lib/Doctrine/DBAL/Driver/IBMDB2/DB2Driver.php

```diff
@@ -14,27 +14,16 @@ class DB2Driver extends AbstractDB2Driver
      */
     public function connect(array $params, $username = null, $password = null, array $driverOptions = [])
     {
-        if (! isset($params['protocol'])) {
-            $params['protocol'] = 'TCPIP';
-        }
+        $params['user']     = $username;
+        $params['password'] = $password;
```

Comment:
User credentials and driver options are passed to the driver both via individual parameters and as part of `$params` which is awkward:
https://github.com/doctrine/dbal/blob/48625f1bc761f1f8f70fe6630a0f3a3204d9a6ff/src/Connection.php#L352-L356

This can be improved by simplifying the signature of `Driver::connect()` to just `Driver::connect($params)` (a separate issue is to be filed). Once it's done, the lines above will be removed.